src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java

```diff
@@ -0,0 +1,304 @@
+     */
+    public static final class DeliveryFactory implements IRequestFactory<Delivery, StandardRequests.DeliveryRequest>
+    {
+
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
+        /**
+         * Moving the curly braces really makes the code hard to read.
+         */
+        public TypeToken<StandardRequests.DeliveryRequest> getFactoryOutputType()
+        {
+            return new TypeToken<StandardRequests.DeliveryRequest>() {};
+        }
+
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
+        /**
+         * Moving the curly braces really makes the code hard to read.
+         */
+        public TypeToken<Delivery> getFactoryInputType()
+        {
+            return new TypeToken<Delivery>() {};
+        }
+
+        /**
+         * Method to serialize a given constructable.
+         *
+         * @param controller The controller that can be used to serialize complicated types.
+         * @param request    The request to serialize.
+         * @return The serialized data of the given requets.
+         */
+        @NotNull
+        @Override
+        public NBTTagCompound serialize(@NotNull final IFactoryController controller, @NotNull final StandardRequests.DeliveryRequest request)
+        {
+            final NBTTagCompound compound = new NBTTagCompound();
+
+            final NBTTagCompound tokenCompound = controller.serialize(request.getToken());
+            final NBTTagInt stateCompound = request.getState().serializeNBT();
+            final NBTTagCompound requestedCompound = request.getRequest().serialize(controller);
+
+            final NBTTagList childrenCompound = new NBTTagList();
+            for (final IToken token : request.getChildren())
+            {
+                childrenCompound.appendTag(controller.serialize(token));
+            }
+
+            compound.setTag(NBT_TOKEN, tokenCompound);
+            compound.setTag(NBT_STATE, stateCompound);
+            compound.setTag(NBT_REQUESTED, requestedCompound);
+
+            if (request.hasResult())
+            {
+                compound.setTag(NBT_RESULT, request.getResult().serialize(controller));
+            }
+
+            if (request.hasParent())
+            {
+                compound.setTag(NBT_PARENT, controller.serialize(request.getParent()));
+            }
+
+            compound.setTag(NBT_CHILDREN, childrenCompound);
+
+            return compound;
+        }
+
+        /**
+         * Method to deserialize a given constructable.
+         *
+         * @param controller The controller that can be used to deserialize complicated types.
+         * @param nbt        The data of the request that should be deserialized.
+         * @return The request that corresponds with the given data in the nbt
+         */
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:LeftCurlyBraceStartLineCheck' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)